source/features/pr-filters.tsx

```diff
@@ -64,25 +68,35 @@ async function addStatusFilter(): Promise<void> {
 		return;
 	}
 
-	// TODO: replace this with an API call
-	const hasCI = await fetchCIStatus();
+	const cachedCIStatus = cache.function(async () => {
+		// TODO: replace this with an API call
+		const statusIcon = await fetchCIStatus();
+		return statusIcon !== undefined;
+	}, {
+		expiration: 3,
+		cacheKey: () => __featureName__ + ':' + getRepoURL()
+	});
```

Comment:
This function needs to be moved to the top-level scope. Best yet if we update it to use the API, example:

```gql
{
  repository(owner: "sindresorhus", name: "refined-github") {
    head: object(expression: "HEAD") {
      ... on Commit {
        history(first: 10) {
          nodes {
            status {
              state
            }
          }
        }
      }
    }
  }
}

```

```js
return repository.head.history.nodes.some((commit: AnyObject) => commit.status);
```

Reply:
I'll have a look at this, so I reverted the cache commit to get this PR ready to merge.